### ISSUE TITLE
[BUG] Building stable release of HO 8.1 failed

### ISSUE THREAD

**Issue by wsbrenk** [ho-dev/HattrickOrganizer#2116]
**Describe the bug**
[A clear and concise description of what the bug is.
](https://github.com/ho-dev/HattrickOrganizer/actions/runs/10021242345/job/27699682506)

> Task :buildSrc:compileKotlin
w: file:///home/runner/work/HattrickOrganizer/HattrickOrganizer/buildSrc/src/main/kotlin/poeditor/PullRunnable.kt:59:50 Unnecessary non-null assertion (!!) on a non-null receiver of type File?
w: file:///home/runner/work/HattrickOrganizer/HattrickOrganizer/buildSrc/src/main/kotlin/poeditor/PullTask.kt:51:50 Type mismatch: inferred type is String? but Any was expected
w: file:///home/runner/work/HattrickOrganizer/HattrickOrganizer/buildSrc/src/main/kotlin/poeditor/PullTask.kt:52:40 Type mismatch: inferred type is List<String>? but (Mutable)Iterable<String?> was expected

> Task :buildSrc:compileJava NO-SOURCE
> Task :buildSrc:compileGroovy NO-SOURCE
> Task :buildSrc:classes UP-TO-DATE

> Task :buildSrc:jar
:jar: No valid plugin descriptors were found in META-INF/gradle-plugins

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring root project 'HO'.
> Could not resolve all files for configuration ':classpath'.
   > Could not find com.overzealous:remark:1.1.0.
Deprecated Gradle features were used in this build, making it incompatible with Gradle 9.0.
     Searched in the following locations:
       - https://plugins.gradle.org/m2/com/overzealous/remark/1.1.0/remark-1.1.0.pom
     Required by:
         project : > org.kordamp.gradle.markdown:org.kordamp.gradle.markdown.gradle.plugin:2.2.0 > org.kordamp.gradle:markdown-gradle-plugin:2.2.0

**Comment by wsbrenk** [comment]
@tychobrailleur Again a poeditor issue!? what do you think about bringing the language translation files back to the HO repo?

**Comment by wsbrenk** [comment]
@tychobrailleur it looks like kotlin supports md to html conversion. it might be the easiest solution to create again our own gradle plugin?

**Comment by tychobrailleur** [comment]
> @tychobrailleur it looks like kotlin supports md to html conversion. it might be the easiest solution to create again our own gradle plugin?

Yeah, as a stop-gap measure, it sounds good.  Let me look into this.

**Comment by tychobrailleur** [comment]
It seems @simonpoole already did the [heavy lift](https://github.com/simonpoole/markdown-gradle-plugin/commit/b1bb3e55b9f0fb9708c8d898e36a6ae5c2bf5cf9) for us; we just have to wait for the 0.2.5 release of the hard fork! ;-)

**Comment by simonpoole** [comment]
Just as a clarification: I hard forked the dependency causing the issue (remark-java) here https://github.com/simonpoole/remark-java, no heavy lifting required (I did update a couple of dependencies and moved the build to gradle). Both the docs and a couple of other things need clean up, but outside of that it works..

The other thing is that I use it in a version of the markdown gradle plugin that I forked a long time ago which has a couple of extra features but which is not synced with the original. Currently I would not recommend using my fork, as there are a number of security issues with the version of flexmark that I'm using (that are however not relevant for my use case) and likely it would probably be better to merge my changes back in to upstream now that it is showing life again.

tl;dr you should get the maintainer of the gradle plugin to use my fork of remark-java if you want quick resolution.

**Comment by tychobrailleur** [comment]
> tl;dr you should get the maintainer of the gradle plugin to use my fork of remark-java if you want quick resolution.

Awesome, thanks for taking the time to clarify the state of your fork, really appreciate!  Will see how to get this addressed upstream.

**Comment by tychobrailleur** [comment]
This [job](https://github.com/ho-dev/HattrickOrganizer/actions/runs/10383327494/job/28748030067) completed successfully.  I'll cherry-pick the change on master.

**Comment by tychobrailleur** [comment]
> This [job](https://github.com/ho-dev/HattrickOrganizer/actions/runs/10383327494/job/28748030067) completed successfully. I'll cherry-pick the change on master.

This is now done, and the [job](https://github.com/ho-dev/HattrickOrganizer/actions/runs/10383458257) completed successfully.
